syntax = "proto3";

package workers_management_service;

option go_package = "github.com/example/workers-management-service/proto";

service WorkersManagementService {
  rpc CreateWorker (CreateWorkerRequest) returns (CreateWorkerResponse);
  rpc GetWorker (GetWorkerRequest) returns (GetWorkerResponse);
  rpc UpdateWorker (UpdateWorkerRequest) returns (UpdateWorkerResponse);
  rpc DeleteWorker (DeleteWorkerRequest) returns (DeleteWorkerResponse);
  rpc ListWorkers (ListWorkersRequest) returns (ListWorkersResponse);
}

message CreateWorkerRequest {
  string name = 1;
  string email = 2;
  string phone_number = 3;
  string department = 4;
  string job_title = 5;
}

message CreateWorkerResponse {
  string id = 1;
  string message = 2;
}

message GetWorkerRequest {
  string id = 1;
}

message GetWorkerResponse {
  string id = 1;
  string name = 2;
  string email = 3;
  string phone_number = 4;
  string department = 5;
  string job_title = 6;
}

message UpdateWorkerRequest {
  string id = 1;
  string name = 2;
  string email = 3;
  string phone_number = 4;
  string department = 5;
  string job_title = 6;
}

message UpdateWorkerResponse {
  string id = 1;
  string message = 2;
}

message DeleteWorkerRequest {
  string id = 1;
}

message DeleteWorkerResponse {
  string message = 1;
}

message ListWorkersRequest {
}

message ListWorkersResponse {
  repeated Worker workers = 1;
}

message Worker {
  string id = 1;
  string name = 2;
  string email = 3;
  string phone_number = 4;
  string department = 5;
  string job_title = 6;
}



// python -m grpc_tools.protoc -I./proto --python_out=server --pyi_out=server --grpc_python_out=server ./proto/workers.proto